distribution/pull.go

```diff
@@ -165,11 +168,11 @@ func Pull(ref reference.Named, imagePullConfig *ImagePullConfig) error {
 // status message indicates that a newer image was downloaded. Otherwise, it
 // indicates that the image is up to date. requestedTag is the tag the message
 // will refer to.
-func writeStatus(requestedTag string, out io.Writer, sf *streamformatter.StreamFormatter, layersDownloaded bool) {
+func writeStatus(requestedTag string, out chan<- xfer.Progress, layersDownloaded bool) {
 	if layersDownloaded {
-		out.Write(sf.FormatStatus("", "Status: Downloaded newer image for %s", requestedTag))
+		out <- xfer.Progress{Message: "Status: Downloaded newer image for " + requestedTag}
 	} else {
-		out.Write(sf.FormatStatus("", "Status: Image is up to date for %s", requestedTag))
+		out <- xfer.Progress{Message: "Status: Image is up to date for " + requestedTag}
```

Comment:
Yes, these channels are buffered. Currently the buffer size is 100 items.

The problem with doing a non-blocking write is that I don't think we want to lose messages if the client can't keep up. Incremental progress messages are unimportant, but messages like this one should not be lost.

Maybe it would make sense to use non-blocking writes in the progress reader (except when it reaches 100%), but use blocking writes for informational messages like this one.